spock-core/src/main/java/org/spockframework/runtime/model/ParameterInfo.java

```diff
@@ -1,11 +1,46 @@
 package org.spockframework.runtime.model;
 
 import java.lang.reflect.Parameter;
+import java.util.Objects;
 
 public class ParameterInfo extends NodeInfo<MethodInfo, Parameter> {
-  public ParameterInfo(MethodInfo parent, String parameterName, Parameter parameter) {
+
+  private int index;
+
+  public ParameterInfo(MethodInfo parent, String parameterName, Parameter parameter, int index) {
     setParent(parent);
     setName(parameterName);
     setReflection(parameter);
+    this.index = index;
```

Comment:
I wonder, if anything (e.g. any extension, I'm not sure if anyone outside could create that object?) could be broken at runtime due to the changed constructor?  If yes, maybe we should keep also the old constructor as deprecated with `-1` used as index?

Reply:
What would you suggest the extensions do when they encounter `ParameterInfo` with `-1`?